lib/pbench/server/unpack_tarballs.py

```diff
@@ -100,34 +65,39 @@ def unpack_tarballs(self, min_size: float, max_size: float) -> Results:
         Returns:
             Results tuple containing the counts of Total and Successful tarballs.
         """
-        tarlist = [
-            tarball
-            for tarball in self.config.ARCHIVE.glob(f"*/{self.LINKSRC}/*.tar.xz")
-            if min_size <= Path(tarball).stat().st_size < max_size
-        ]
-
-        ntotal = nsuccess = 0
-        file_tree = FileTree(self.config, self.logger)
-
-        for tarball in sorted(tarlist):
-            ntotal += 1
-
-            try:
+        datasets = self.sync.next(Operation.UNPACK)
+        tarlist: list[Target] = []
+        for d in datasets:
+            t = Metadata.getvalue(d, Metadata.TARBALL_PATH)
+            if t:
                 try:
-                    tb = Path(tarball).resolve(strict=True)
+                    p = Path(t).resolve(strict=True)
+                    s = p.stat().st_size
                 except FileNotFoundError as exc:
-                    self.logger.error(
-                        "{}: Tarball link, '{}', does not resolve to a real location: {}",
+                    self.logger.exception(
+                        "{}: Tarball '{}' does not resolve to a file: {}",
                         self.config.TS,
-                        tarball,
+                        t,
                         exc,
                     )
-                    raise
-                self.unpack(tb, file_tree)
-                self.update_symlink(tb)
-            except Exception:
-                continue
+                    continue
+                except Exception:
+                    self.logger.exception("Unexpected exception on {}", t)
+                    continue
+
+                if min_size <= s < max_size:
+                    tarlist.append(Target(d, p))
```

Comment:
Since we can, it might be nice to use keyword arguments in the c'tor call:
```
                    tarlist.append(Target(dataset=d, tarball=p))
```